pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/RoundDecimalTransformFunction.java

```diff
@@ -90,18 +90,33 @@ public double[] transformToDoubleValuesSV(ValueBlock valueBlock) {
     double[] leftValues = _leftTransformFunction.transformToDoubleValuesSV(valueBlock);
     if (_fixedScale) {
       for (int i = 0; i < length; i++) {
-        _doubleValuesSV[i] = BigDecimal.valueOf(leftValues[i])
-            .setScale(_scale, RoundingMode.HALF_UP).doubleValue();
+        if (leftValues[i] == Double.NEGATIVE_INFINITY || leftValues[i] == Double.POSITIVE_INFINITY ||
+                leftValues[i] == Double.NaN) {
```

Comment:
`NaN` is not comparable. We can probably use a try-catch over result computation, and set the result to be left value if exception is caught

Reply:
Or we can check with Double.isNaN. It's documentation says:
'This method corresponds to the isNaN operation defined in IEEE 754.'

Reply:
Sounds good. Let's follow the same behavior then. You may use a try-catch to avoid the overhead of value check for happy path

Reply:
Currently Double.NaN, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY are put into null bit map. I am debugging code to ensure that these 3 values are not part of null bit map.